### PR TITLE
Add DisableIntegrityMonitoring parameter to New-AzVM and change default behavior for a particular TrustedLaunch scenario

### PR DESCRIPTION
**Pull request by Sandido**
## Description

This PR modifies the default behavior when this scenario occurs:
1) SecurityType on the VM is TrustedLaunch.
2) VTpmEnabled is true.
3) SecureBootEnabled is true. 
4) DisableIntegrityMonitoring (the new switch parameter) is not true. 
When this scenario occurs, the VM Extension 'Guest Attestation' is added to the new VM, and the VM Identity will be set to SystemAssigned. 

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1166 
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Review by VeryEarly — DISMISSED** [review]

**Comment by Sandido** [comment]
@VeryEarly , not sure if these CI checks were failing before I added the vmss work, but the test failures look like a framework issue not something due to this pr. 
I even made another PR reverted to what you previously reviewed and the same errors are there: https://github.com/Azure/azure-powershell/pull/18763

**Comment by Sandido** [comment]
I'm unable to run the tests locally due to a known issue, so please ensure this doesn't break anything else before merging.

File: src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs [hunk 1]
```diff
@@ -897,11 +914,67 @@ public void DefaultExecuteCmdlet()
...
                 });
             }
         }
 
+        /// Summary: Method to determine if the Guest Attestation extension
+        /// should be installed on the VM being created.
+        /// Currently only functions with the DefaultParamSet parameter set. 
+        private bool shouldGuestAttestationExtBeInstalled()
+        {
+            if (this.DisableIntegrityMonitoring != true &&
```
Comment (VeryEarly) [inline src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs:965]: test error thrown at this line

**Review by VeryEarly — CHANGES_REQUESTED** [review]

**Review by VeryEarly — APPROVED** [review]

**Pull request merged**